supersuit/vector/markov_vector_wrapper.py

```diff
@@ -91,9 +91,14 @@ def step(self, actions):
         infs = [infos.get(agent, {}) for agent in self.par_env.possible_agents]
 
         if env_done:
-            observations, infs = self.reset()
+            observations, reset_infs = self.reset()
         else:
             observations = self.concat_obs(observations)
+            # empty infos for reset infs
+            reset_infs = [{} for _ in range(len(self.par_env.possible_agents))]
+        # combine standard infos and reset infos
+        infs = infs + reset_infs
```

Comment:
Looks good to me, I think the name change is good as well

Reply:
Could you add a test to check that this works correctly. I'm confused on what this changes practically

Reply:
Looks like it makes the info length twelve instead of six?https://github.com/Farama-Foundation/SuperSuit/actions/runs/6696953379/job/18341926381?pr=233

Reply:
Probably a normal dict join would be better. Something like:

```
infs = [{**inf, **reset_inf} for inf, reset_inf in zip(infs, reset_infs)]
```

Reply:
So I have updated the code following @benblack769 suggestion (that should fix the length error @elliottower since know each agent has just infos combining normal and reset info ) and added a test to check if this works.

Reply:
Looking at the CI, it seems to raise an error with the test

> FAILED test/test_vector/test_pettingzoo_to_vec.py::test_terminal_obs_are_returned - AssertionError: MarkovVectorEnv does not support environments with varying numbers of active agents unless black_death is set to True

Reply:
So looks like there was a change around 2 months ago in [pettingzoo's tests](https://github.com/Farama-Foundation/PettingZoo/blame/c65e8947f2b20164dd3bea0a300d42ee9b705594/pettingzoo/utils/wrappers/terminate_illegal.py#L34) that expects infos to be dict with agent id's as keys. 

This is inconsistent with how supersuit handles the infos (list of dicts, with no keys). I reverted my changes locally and the tests still fail with the latest version of supersuit and pettingzoo.

Reply:
Oh good catch, yeah with parallel environments it's expected in other wrappers internally for conversions that the info is indexed by agent ID. It was implicitly a thing you were supposed to do previously because otherwise the wrappers wouldn't work and other things assumed it was done correctly, but we made it strict just to make sure things all made sense and worked the way they're supposed to.

Anyways, trying to debug this locally and it looks like at least the `test_nan_random` one fails because the environment has no agents and it tries to observe--not sure what would cause this. It looks like the other tests also fail with the same reason it seems.

Reply:
Okay I fixed the issue, just made it so the terminate illegal wrapper checks first to see if the agent's name key is in the `self.infos` dict, otherwise it sets the prev_info to {}